src/app/qgisapp.cpp

```diff
@@ -16068,6 +16068,7 @@ void QgisApp::write3DMapViewSettings( Qgs3DMapCanvasWidget *widget, QDomDocument
   QDomElement elemCamera = widget->mapCanvas3D()->cameraController()->writeXml( doc );
   elem3DMap.appendChild( elemCamera );
   QDomElement elemAnimation = widget->animationWidget()->animation().writeXml( doc );
+  elemAnimation.setAttribute( QStringLiteral( "widget-visible" ), widget->animationWidget()->isVisible() ? 1 : 0 );
```

Comment:
The logic for QWidget::isVisible is actually quite odd. It will return false if the parent widget is hidden too -- so eg a widget in a dialog which isn't visible will return false to isVisible, even if the widget hasn't been explicitly hidden at all. It's better to use isHidden instead, which returns true ONLY if the widget itself has been explicitly hidden, and which ignores the parent widget visibility entirely.
```suggestion
  elemAnimation.setAttribute( QStringLiteral( "widget-visible" ), !widget->animationWidget()->isHidden() ? 1 : 0 );
```

Reply:
Done.